engine/execution/computation/computer/computer.go

```diff
@@ -245,12 +245,13 @@ func (e *blockComputer) executeSystemCollection(
 	if err != nil {
 		return txIndex, err
 	}
-	if res.TransactionResults[len(res.TransactionResults)-1].ErrorMessage != "" {
+	systemChunkTxResult := res.TransactionResults[len(res.TransactionResults)-1]
```

Comment:
I'm a bit worried - `txIndex` should correspond to index in `TransactionResults`, its 0 based as well. 
Does it indicate maybe some other part if wrong?

Reply:
I think it does when we call `executeTransaction`, but then it gets [incremented afterward](https://github.com/onflow/flow-go/pull/1125/files#diff-94b6f84058d4bc409d139e005e5e484de3d76dab187c7c4f71292178dafd8e76R243) which seems like the problem. 

It seems like it's returning the `txIndex` for the next transaction to be executed, not the last transaction that was executed.